lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -351,11 +355,19 @@ void FIRRTLModuleLowering::runOnOperation() {
       continue;
     }
 
-    // Otherwise we don't know what this is.  We are just going to drop it,
-    // but emit an error so the client has some chance to know that this is
-    // going to happen.
-    op.emitError("unexpected operation '")
-        << op.getName() << "' in a firrtl.circuit";
+    // Anything which is _not_ a module or an extmoulde is treated carefully.
+    // By default, this produces an error.
+    TypeSwitch<Operation *>(&op)
+        // GrandCentral can generate interfaces.  These need to be let through.
+        .Case<sv::InterfaceOp>(
+            [&](auto op) { state.passThroughOps.push_back(op); })
+        .Default([](auto op) {
+          // Otherwise we don't know what this is.  We are just going to drop
+          // it, but emit an error so the client has some chance to know that
+          // this is going to happen.
+          op->emitError("unexpected operation '")
+              << op->getName() << "' in a firrtl.circuit";
+        });
```

Comment:
I like the white-listing approach here. That way we can carefully control what gets passed through. Maybe there's some point to be made that in the MLIR spirit unknown ops should just be passed through anyway, but that seems to make the interface more leaky and unpredictable. I like this approach.

Reply:
Cool.  I figured this was safer to start letting ops through on a case-by-case basic for now.  (Also, it's a smaller delta for this patch series to just let one thing through.)  I totally agree about the MLIR spirit, though.  

It probably makes sense to remove the check in a subsequent patch.  I can see third-party FIRRTL Dialect pass writers needing to let anything through.